pipe/stage.go

```diff
@@ -32,3 +108,54 @@ type Stage interface {
 	// the context passed to `Start()`.
 	Wait() error
 }
+
+// StagePreferences is the way that a Stage2 indicates its preferences
+// about how it is run. This is used within `pipe.Pipeline` to decide
+// when to use `os.Pipe()` vs. `io.Pipe()` for creating the pipes
+// between stages.
+type StagePreferences struct {
+	StdinPreference  IOPreference
+	StdoutPreference IOPreference
+}
+
+// Stage2 is a `Stage` that can accept both stdin and stdout arguments
+// when it is started.
+type Stage2 interface {
```

Comment:
`Stage2` sounds like something that runs after `Stage1`, rather than something that extends `Stage`. This is probably because of bootloaders and such, which is not particularly relevant here. Is there a pattern of suffixing numbers in Go? Would we rather go with something like `StageWithIO` so you then have `StartWithIO()`? That name's not amazing either, but it might still be worth capturing *what* is different about this interface.

Reply:
That's a good point. I'll rename the interface (or feel free to do so yourself if you are in a hurry).

Reply:
I just pushed the rename that you suggested.